Framework/Algorithms/test/BinaryOperationTest.h

```diff
@@ -174,18 +175,14 @@ class BinaryOperationTest : public CxxTest::TestSuite {
     TS_ASSERT(output);
 
     for (int i = 0; i < nHist; ++i) {
-      IDetector_const_sptr det;
-      try {
-        det = output->getDetector(i);
-      } catch (Mantid::Kernel::Exception::NotFoundError &) {
-      }
+      auto spectrumInfo = output->spectrumInfo();
 
-      TSM_ASSERT("Detector was found", det);
-      if (det) {
+      TSM_ASSERT("Detector was found", spectrumInfo.hasDetectors(i));
```

Comment:
I appreciate that this was like this already, but when the assertion failes the user will see the message "Detector was found" implying that it should not have been. You need to introduce a negative into the test message.